SimMuon/MCTruth/python/NewMuonAssociatorByHits_cfi.py

```diff
@@ -0,0 +1,147 @@
+import FWCore.ParameterSet.Config as cms
```

Comment:
Maybe you forgot what is the strategy of these develoments. It was explained long time ago with the first PR:
https://github.com/cms-sw/cmssw/pull/15153

The prefix "New" is there as an easy-to-see flag, as the step we are making now is just a temporary one. The "New" code will coexist with the production code (without the prefix "New") just for a short time. This time is necessary to validate the new validation code, which is changing very much. This is also why we wanted to make this validation of the code on a stable release (90X) to decouple the validation of the code from the standard release validation. 
After the validation is done the new code will replace the old one and all the prefixes "New" disappear. There will be no duplication of code and no ambiguity.

About the time/memory performance, it does not change substantially (but it does not worsen anything). The code readibility improves a lot. Improving the technical performance is planned for the next developments, after the current plan is accomplished.